docs/modules/ROOT/pages/servlet/oauth2/client/client-authentication.adoc

```diff
@@ -92,7 +92,9 @@ val tokenResponseClient = DefaultAuthorizationCodeTokenResponseClient()
 tokenResponseClient.setRequestEntityConverter(requestEntityConverter)
 ----
 ======
-
+[NOTE]
+If you're using the `client-authentication-method: client_secret_basic` and you need to skip URL encoding,
+create a new `DefaultOAuth2TokenRequestHeadersConverter` and set it in the Request Entity Converter above.
```

Comment:
This note seems to be added in the "Authenticate using `private_key_jwt" section? If so, this wouldn't be the best place to add this note. Perhaps it could be a `[TIP]` elsewhere? For example, it could be added to `authorization-grants.adoc` where we cover customizing the access token request for each grant type. But this would require adding the tip 5 times because we unfortunately have structured the content that way currently. Or we could (for now) add an "Authenticate using `client_secret_basic`" section on this page.

Reply:
I'll format and push the code, and we can discuss the documentation side after we feel good about the integration of the `DefaultOAuthtokenREquestHeadersConverter` into the Reactive and Non-Reactive side. Since I can see that impacting the documentation we set up.

Reply:
Let's fix things so that tests are not broken (see above recommendations). As mentioned above, let's also remove `getDefaultTokenRequestHeaders()`.